torch_geometric/nn/conv/message_passing.py

```diff
@@ -112,14 +109,16 @@ def __init__(self, aggr: Optional[Union[str, List[str]]] = "add",
 
         self.inspector = Inspector(self)
         self.inspector.inspect(self.message)
+        self.inspector.inspect(self.explain_message, pop_first=True)
```

Comment:
I think we only want to do this in case `self._explain` is `True`. Otherwise, the arguments of `explain_message` will get collected even if the function is never called. I think we can achieve this by adding a `requires_explain(True)` function to `MessagePassing`, and do inspecting and adding to `__user_args__` solely there. WDYT?